## Hugging face embedding model usage

In [1]:
from langchain_huggingface import HuggingFaceEmbeddings

## Initialise a simple Embedding model (no key needed)

embeddings = HuggingFaceEmbeddings(
    model_name="sentence-transformers/all-MiniLM-L6-v2"
)

embeddings

HuggingFaceEmbeddings(model_name='sentence-transformers/all-MiniLM-L6-v2', cache_folder=None, model_kwargs={}, encode_kwargs={}, query_encode_kwargs={}, multi_process=False, show_progress=False)

In [6]:
## create first embeddings
text="Hello! I am learning about embeddings"

embedding = embeddings.embed_query(text)

print(f"len {len(embedding)}")

documents = ["The cat sat on bank","What is going on?","Am I dreaming?"]

embeddingDoc = embeddings.embed_documents(documents)
print(f"Len of doc {len(embeddingDoc)} each with vector size of {len(embeddingDoc[0])}")

len 384
Len of doc 3 each with vector size of 384


### Popular Open Source Embeddings Models:

Model: sentence-transformers/all-MiniLM-L6-v2
- 384 dimensions
- fast efficient, good quality
- use: General Purpose, real time applications

Model: sentence-transformers/all-mpnet-base-2
- 768 dimensions
- best quality,  slower than MiniLM
- use: quality over speed

Model: sentence-transformers/all-MiniLM-L12-v2
- 384 dimensions
- slightly better than L6, bit slower
- use: good in speed and quality
    
Model: sentence-transformers/multi-qa-MiniLM-L6-cos-v1
- 384 dimensions
- optimized for qna
- use: Q&A , sementic search
    
Model: sentence-transformers/paraphrase-multilingual-MiniLM-L12-v2
- 384 dimensions
- supports 50+ lang
- use: Multilingual application